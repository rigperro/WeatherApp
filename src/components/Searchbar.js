import React, { useState } from 'react';
import styled from 'styled-components';

// This component is simply a text input field with a search button
// that calls the parent components getWeather
const Searchbar = ({getWeather}) => {
  const [input, setInput] = useState('');

  return(
    <form onSubmit = {event => {
      event.preventDefault();
      if (input) {
        getWeather(input);
        setInput('');
      };
    }}>
        <StyledInput
          value={input}
          type="text"
          placeholder={'Enter city to search'}
          name="city"
          onChange={event => setInput(event.target.value)}
        />
        <StyledButton type="submit">
          Go
        </StyledButton>
    </form>
  );
};

const StyledInput = styled.input`
  border: none;
  border-radius: 0;
  border-bottom: 2px solid white;
  background: transparent;
  padding: 12px 20px;
  font-size: calc(10px + 2vmin);
  color: white;
`

const StyledButton = styled.button`
  background: transparent;
  border: 3px solid white;
  border-radius: 3px;
  color: white;
  margin: 0.5em 1em;
  padding: 0.4em 0.8em;
  &:hover {
  background: #260A13;
  }
`

export default Searchbar;
