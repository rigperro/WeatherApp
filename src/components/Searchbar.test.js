import React from 'react';
import ReactDOM from 'react-dom';
import Searchbar from './Searchbar';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

it('renders without chrashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Searchbar />, div);
});

test('Searchbar renders correctly', () => {
  const tree = renderer
    .create(<Searchbar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
