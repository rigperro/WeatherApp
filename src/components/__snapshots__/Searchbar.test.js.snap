// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders correctly 1`] = `
.c0 {
  border: none;
  border-radius: 0;
  border-bottom: 2px solid white;
  background: transparent;
  padding: 12px 20px;
  font-family: 'Oxygen',sans-serif;
  font-size: calc(10px + 2vmin);
  color: white;
}

.c1 {
  background: transparent;
  border: 3px solid white;
  border-radius: 3px;
  color: white;
  margin: 0.5em 1em;
  font-family: 'Oxygen',sans-serif;
  padding: 0.4em 0.8em;
}

.c1:hover {
  background: #260A13;
}

<form
  onSubmit={[Function]}
>
  <input
    className="c0"
    name="city"
    onChange={[Function]}
    placeholder="Enter city to search"
    type="text"
    value=""
  />
  <button
    className="c1"
    type="submit"
  >
    Go
  </button>
</form>
`;
